### PR TITLE
codegen: rewrite ztypes parser in nom

### PR DESCRIPTION
**Pull request by bouk**
* Parses nested structs
* Should be easier to extend

I wasn't able to run the codegen because my dev setup is on macOS...

Related to #13

### CODE REVIEW

File: codegen/src/generator/ztypes.rs [hunk 1]
```diff
@@ -1,113 +1,58 @@
...
-        "a" => translate_container(rest),
-        "(" => translate_tuple_body(rest),
-        x => bail!("Unexpect starting character '{}'", x),
-    }
+fn parse_array(input: &str) -> IResult<&str, String> {
+    map(preceded(tag("a"), parse_type), |t| format!("Vec<{}>", t)).parse(input)
 }
 
-pub(crate) fn translate_container(sig: &str) -> Result<(String, &str)> {
+pub fn translate_sig(sig: &str) -> Result<String> {
```
Comment (lucab) [inline codegen/src/generator/ztypes.rs:47]: Minor: can you please retain the `#[context(...)]` that was on this, and move the function to the top of the file?

**Comment by bouk** [comment]
Added back the context

**Comment by lucab** [comment]
Thanks for the PR, it looks good to me and I manually verified that there are no codegen regressions.

Could you please rebase against current `main` and re-run the code generator? I did a systemd bump to `v253` in the meanwhile, so it would be better to refresh the generated code here.

As a sidenote, if you have https://github.com/casey/just installed locally then the whole refresh can be performed through `just bump-systemd && just default`.

**Comment by bouk** [comment]
Oh cool I'll add a note about `just`!

**Comment by bouk** [comment]
@lucab I've implement parsing dictionary signatures as well, which makes the parser complete!

**Review by lucab — APPROVED** [review]

**Pull request merged**